src/Reopt/CFG/FnRep.hs

```diff
@@ -234,11 +234,18 @@ data FnValue (arch :: Type) (tp :: M.Type) where
     -- | Symbol name of this function.
     BSC.ByteString ->
     FnValue arch (BVType (ArchAddrWidth arch))
-  -- | Value is a function.
-  --
-  -- The int should be in the range @[0..argCount)@, and the type repr
-  -- is the type of the argument.
+  -- | Value is a function argument.  The int should be in the range
+  -- @[0..argCount)@, and the type repr is the type of the argument.
   FnArg :: !Int -> !(TypeRepr tp) -> FnValue arch tp
+  -- | Value is a constant pointer into the executable segment
+  FnCodePointer ::
+    Macaw.MemAddr (ArchAddrWidth arch) ->
+    FnValue arch (BVType (ArchAddrWidth arch))
+  -- | Value is a constant pointer into the executable segment
+  FnTypedCodePointer ::
```

Comment:
These two constructors have the same Haddocks... say something about what makes them different?